cli/src/worker_pool.js

```diff
@@ -0,0 +1,103 @@
+import { Worker, parentPort } from "worker_threads";
+import { TransformStream } from "web-streams-polyfill";
+
+function uuid() {
+  return Array.from({ length: 16 }, () =>
+    Math.floor(Math.random() * 256).toString(16)
+  ).join("");
+}
+
+function jobPromise(worker, msg) {
+  return new Promise(resolve => {
+    const id = uuid();
+    worker.postMessage({ msg, id });
+    worker.on("message", function f({ result, id: rid }) {
```

Comment:
Should worker errors be handled as well? Or are they handled elsewhere?

Reply:
The error handling of the entire CLI is not good. But I could at least add a log message I guess.